From be542652d1c8fac1e706e4bffbb46d4814051467 Mon Sep 17 00:00:00 2001
From: Frank Li <Frank.Li@freescale.com>
Date: Fri, 7 Feb 2014 06:23:03 +0800
Subject: [PATCH 1038/1074] ENGR00299323-19 net:fec: fix crash when no
 ethernet card attached

Fix kernel crash when no ethernet card attached.

Signed-off-by: Frank Li <Frank.Li@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 drivers/net/ethernet/freescale/fec_main.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/freescale/fec_main.c b/drivers/net/ethernet/freescale/fec_main.c
index fd95aad..fdc4950 100644
--- a/drivers/net/ethernet/freescale/fec_main.c
+++ b/drivers/net/ethernet/freescale/fec_main.c
@@ -2085,6 +2085,9 @@ static int fec_enet_alloc_buffers(struct net_device *ndev)
 }
 
 static int
+fec_enet_close(struct net_device *ndev);
+
+static int
 fec_enet_open(struct net_device *ndev)
 {
 	struct fec_enet_private *fep = netdev_priv(ndev);
@@ -2101,7 +2104,7 @@ fec_enet_open(struct net_device *ndev)
 	/* Probe and connect to PHY when open the interface */
 	ret = fec_enet_mii_probe(ndev);
 	if (ret) {
-		fec_enet_free_buffers(ndev);
+		fec_enet_close(ndev);
 		return ret;
 	}
 
-- 
1.7.5.4

